### DISCUSSION TITLE
why 5 processes in background?

### DISCUSSION THREAD

**Discussion by bmw216** [xanderfrangos/twinkle-tray#866] [Q&A]
why 5 processes in background?

![Skärmbild 2024-10-22 221726](https://github.com/user-attachments/assets/24b5250a-127e-453e-a248-f030c2a4f6fa)

**Comment by xanderfrangos** [comment]
So in order...

![Screenshot 2024-10-24 192429](https://github.com/user-attachments/assets/ed96fe18-f59e-4273-9848-27e5947b3b8d)

1. The main process
2. A process that handles GPU memory for the render.
3. A process that handles networking and other utility features.
4. The renderer process for the flyout UI.
5. A dedicated process for display communication.

Twinkle Tray is built with Electron, which forces 2, 3, and 4 to exist separately. 1 and 5 are split intentionally, as display communication can momentarily lock the whole process. Meaning the Twinkle Tray UI could freeze while getting new brightness values if they were the same process. In general, these processes aren't consuming any CPU time when you're not actively using Twinkle Tray, so the quantity isn't really an issue.

**Reply by bmw216** [reply]
Its ok. This app is 9 million times better than lg onscreen control. ❤

**Reply by flodavid** [reply]
Thank you for the explanation above @xanderfrangos . Would it be possible to have an option to not launch process 2 and 3 at startup (4 is probably mandatory if I understand correctly), until the "Twinkle Tray Settings" window is open ? Most of the time, we only need the tray icon, to be able to use the mouse wheel on it. It would save some RAM, has the usage is currently 100 Mo when the service starts on my machine.
Great job for this app anyway.
